### ISSUE TITLE
crypto/tls: TestBoringClientHello failure

### ISSUE THREAD

**Issue by derekparker** [golang-fips/go#51]
Testing the latest commit on master I'm now seeing the following failure:

```
--- FAIL: TestBoringClientHello (0.00s)
    boring_test.go:292: client offered disallowed suite 0x1303
FAIL
FAIL    crypto/tls      2.827s
```

**Comment by derekparker** [comment]
I believe this is because we've started enabling TLS 1.3, the disallowed suite causing the test to fail is `TLS_CHACHA20_POLY1305_SHA256`.

cc @ueno

**Comment by derekparker** [comment]
https://github.com/golang/go/blob/release-branch.go1.19/src/crypto/tls/cipher_suites.go#L692

**Comment by ueno** [comment]
Hmm, that's strange, as we exclude `TLS_CHACHA20_POLY1305_SHA256` [here](https://github.com/golang-fips/go/blob/09a449151e2d41b739f831b9e85fa9710cd2beed/patches/000-initial-setup.patch#L454) and [here](https://github.com/golang-fips/go/blob/09a449151e2d41b739f831b9e85fa9710cd2beed/patches/000-initial-setup.patch#L428). I've tried to reproduce with the following, but couldn't.
```console
$ fips-mode-setup --check
FIPS mode is enabled.
$ cd go
# apply patches
$ cd src
$ ./make.bash
$ cd ..
$ export GOROOT=$PWD
$ export PATH="$GOROOT/bin:$PATH"
$ cd src/crypto/tls
$ GOLANG_FIPS=1 go test -v -run "Boring"
=== RUN   TestBoringClientHello
--- PASS: TestBoringClientHello (0.00s)
```

**Comment by derekparker** [comment]
@ueno run it without `GOLANG_FIPS=1` and it should fail.

**Comment by dbenoit17** [comment]
Fixed by #54
